icicle-source/src/Icicle/Source/Checker/Base.hs

```diff
@@ -292,20 +293,20 @@ removeElementBinds env
    in  foldr Map.delete env elts
  where
   isElementTemporality ft
-   = getTemporalityOrPure (functionReturn ft) == TemporalityElement
+   = getTemporalityOrPure ft == TemporalityElement
```

Comment:
Should this strip off the arrows from `ft`? Or maybe `getTemporality` should look down into the return type of arrows and get the temporality there